pkg/git/github_test.go

```diff
@@ -0,0 +1,681 @@
+package git
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/google/go-github/v35/github"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func TestNewGitHubGITProvider(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "GitHubProvider")
+}
+
+var _ = Describe("NewGitHubGITProvider", func() {
+	var err error
+	var ctx context.Context
+
+	BeforeEach(func() {
+		err = nil
+		ctx = context.Background()
+	})
+
+	It("returns error when creating without url", func() {
+		_, err = NewGitHubGITProvider(ctx, "", "foo")
+		Expect(err).To(MatchError("remoteURL empty"))
+	})
+
+	It("returns error when creating without token", func() {
+		_, err = NewGitHubGITProvider(ctx, "https://github.com/org/repo", "")
+		Expect(err).To(MatchError("token empty"))
+	})
+
+	It("returns error when creating without github address", func() {
+		_, err = NewGitHubGITProvider(ctx, "https://foo.bar/org/repo", "foo")
+		Expect(err).To(MatchError("host does not start with https://github.com: https://foo.bar"))
+	})
+
+	It("returns error when creating with fake token", func() {
+		_, err = NewGitHubGITProvider(ctx, "https://github.com/org/repo", "foo")
+		Expect(err).To(MatchError("unable to authenticate using token"))
+	})
+
+	It("is successfully created when creating with correct token", func() {
+		var provider *GitHubGITProvider
+		remoteURL := os.Getenv("GITHUB_URL")
+		token := os.Getenv("GITHUB_TOKEN")
+
+		if remoteURL == "" || token == "" {
+			Skip("GITHUB_URL and/or GITHUB_TOKEN environment variables not set")
+		}
+
+		provider, err = NewGitHubGITProvider(ctx, remoteURL, token)
+		Expect(err).To(BeNil())
+		Expect(remoteURL).To(ContainSubstring(provider.owner))
+		Expect(remoteURL).To(ContainSubstring(provider.repo))
+	})
+})
+
+var _ = Describe("GitHubGITProvider CreatePR", func() {
```

Comment:
Should we have two kind of tests, one unit tests and one e2e where we need a real github token?

Reply:
Yep, but then we need a mocked provider - and I see that as another PR.